daemon/cmd/kube_proxy_replacement.go

```diff
@@ -295,8 +295,9 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		switch {
 		case !option.Config.EnableNodePort:
 			msg = fmt.Sprintf("BPF host routing requires %s.", option.EnableNodePort)
-		case option.Config.Tunnel != option.TunnelDisabled:
-			msg = fmt.Sprintf("BPF host routing is only available in native routing mode.")
+		// If we chain CNIs then all bets are off.
+		case option.Config.IsFlannelMasterDeviceSet():
```

Comment:
JFYI, we have deprecated the Flannel mode, and in the nearest future we should remove once we have validated that    the CNI chaining with Flannel works.

Reply:
Thanks for clarifying, I wasn't quite sure. Removing the flannel specific code would actually simplify the datapath a bit. Maybe I'll make a new PR and remove flannel bits.

Reply:
Maybe we could add that to the list of guide to validate for 1.10 and remove early in the 1.11 cycle?